### PR TITLE
[bitnami/thanos] provide Objstore config via secret

### LINKED ISSUE #2353: [bitnami/thanos]: change ObjectStore configMap into secret
**Which chart**: bitnami/thanos

**Is your feature request related to a problem?**
ObjectStore is defined in a file [objstore-configmap.yaml](https://github.com/bitnami/charts/blob/master/bitnami/thanos/templates/objstore-configmap.yaml), representing a Kubernetes ConfigMap resource.
The same ObjectStore should be passed to thanos-sidecar while configuring prometheus-operator as specified [here](https://github.com/bitnami/charts/tree/master/bitnami/prometheus-operator/) where the tag `prometheus.thanos.objectStorageConfig.secretName` is a Kubernetes Secret resource (not a ConfigMap).
This is a conflicting definition (configMap vs secret) that makes hard to reuse the same information in more places.

**Describe the solution you'd like**
Considering that ObjectStore may contain access and secret key (to access a bucket), it should be a secret.
In Thanos chart it should be considered the idea of transforming all references to configMap into references to a secret and make ObjectStore a secret.

### PR DESCRIPTION
**Pull request by Whyeasy**
**Description of the change**

With this change it is possible to provide the Objstore configuration via a existing secret. It also has the capability to provide keys for you secret, this way we can expand the functionality with for example Tracing config.

Also fixed some typos in the Readme.md

**Benefits**

Keep the configuration and password to your Objstore private.

**Possible drawbacks**

Still expects that you name the path of your secret (existingObjstoreSecret.items) objstore.yaml for the objstore config.

**Applicable issues**

  - fixes #2353

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/thanos/README.md [hunk 1]
```diff
@@ -89,18 +89,20 @@ The following tables lists the configurable parameters of the Thanos chart and t
...
+| `image.pullPolicy`              | Thanos image pull policy                               | `IfNotPresent`                                          |
+| `image.pullSecrets`             | Specify docker-registry secret names as an array       | `[]` (does not add image pull secrets to deployed pods) |
+| `nameOverride`                  | String to partially override thanos.fullname           | `nil`                                                   |
+| `fullnameOverride`              | String to fully override thanos.fullname               | `nil`                                                   |
+| `clusterDomain`                 | Default Kubernetes cluster domain                      | `cluster.local`                                         |
+| `objstoreConfig`                | Objstore configuration                                 | `nil`                                                   |
+| `existingObjstoreConfigmap`     | Name of existing ConfigMap with Objstore configuration | `nil`                                                   |
```
Comment (juan131) [inline bitnami/thanos/README.md:103]: This parameter should be named `existingObjstoreSecret` instead, right? Also the description should state it's the name of an existing Secret instead.

File: bitnami/thanos/templates/_helpers.tpl [hunk 2]
```diff
@@ -363,14 +363,15 @@ Compile all warnings into a single message, and call fail.
...
 {{/* Validate values of Thanos - Objstore configuration */}}
 {{- define "thanos.validateValues.objstore" -}}
-{{- if and (or .Values.bucketweb.enabled .Values.compactor.enabled .Values.ruler.enabled .Values.storegateway.enabled) (not (include "thanos.createObjstoreConfigmap" .)) (not .Values.existingObjstoreConfigmap) -}}
+{{- if and (or .Values.bucketweb.enabled .Values.compactor.enabled .Values.ruler.enabled .Values.storegateway.enabled) (not (include "thanos.createObjstoreConfigmap" .)) (not .Values.existingObjstoreConfigmap) ( not .Values.existingObjstoreSecret) -}}
 thanos: objstore configuration
     When enabling Bucket Web, Compactor, Ruler or Store Gateway component,
     you must provide a valid objstore configuration.
     There are three alternatives to provide it:
       1) Provide it using the 'objstoreConfig' parameter
       2) Provide it using an existing Configmap and using the 'existingObjstoreConfigmap' parameter
```
Comment (juan131) [inline bitnami/thanos/templates/_helpers.tpl:372]: This alternative should not be supported anymore.

File: bitnami/thanos/Chart.yaml [hunk 3]
```diff
@@ -15,4 +15,4 @@ maintainers:
 name: thanos
 sources:
   - https://github.com/bitnami/bitnami-docker-thanos
-version: 0.4.3
+version: 0.4.4
```
Comment (juan131) [inline bitnami/thanos/Chart.yaml:18]: Transforming the Objstore into a secret might break backwards incompatibility. 
Please test upgrading from previous version to the one including your changes to ensure it works smoothly. If that's not the case, we might need to bump the major version to include this change.

File: bitnami/thanos/README.md [hunk 4]
```diff
@@ -89,18 +89,20 @@ The following tables lists the configurable parameters of the Thanos chart and t
...
+| `existingObjstoreSecret.name`   | Name of existing secret with Objstore configuration    | `nil`                                                   |
+| `existingObjstoreSecret.items`  | List of secrets keys to paths                          | `[]`                                                    |
```
Comment (juan131) [inline bitnami/thanos/README.md:105]: Remember to update the section `### Using custom Objstore configuration` of this **README.md**

File: bitnami/thanos/Chart.yaml (hunk 3 above)
Comment (Whyeasy) [reply bitnami/thanos/Chart.yaml]: Upgraded to major release, we are removing a capability now and won't work with older values.yaml.

**Comment by juan131** [comment]
```diff
diff --git a/bitnami/thanos/templates/objstore-secret.yaml b/bitnami/thanos/templates/objstore-secret.yaml
index 1c548c53..c71b07e5 100644
--- a/bitnami/thanos/templates/objstore-secret.yaml
+++ b/bitnami/thanos/templates/objstore-secret.yaml
@@ -6,5 +6,5 @@ metadata:
   labels: {{- include "thanos.labels" . | nindent 4 }}
 data:
   objstore.yml: |-
-    {{- include "thanos.tplValue" (dict "value" .Values.objstoreConfig "context" $) | nindent 4 }}
+    {{- include "thanos.tplValue" (dict "value" .Values.objstoreConfig "context" $) | b64enc | nindent 4 }}
 {{ end }}
```

**Comment by juan131** [comment]
Waiting for CI/CD to test your changes and update image tags

**Pull request merged**